packages/utils/models.js

```diff
@@ -280,15 +363,22 @@ export const NO_PRINTER = [MODELS.MP35, MODELS.D199, MODELS.D195];
  * @returns {boolean}
  */
 export function hasPrinter() {
-  return VerifyMethodOnSystemWrapper('hasPrinter') || !_hasModelAtList(NO_PRINTER);
+  /** negação explicada e comentada em hasNoTouch() */
+  let value = VerifyMethodOnSystemWrapper('hasPrinter');
+
+  if (!value) {
+    value = _hasModelAtList(NO_PRINTER);
+  }
+
+  return !value;
```

Comment:
Esse é difente, porque a gente não tem uma lista `HAS_PRINTER`. temos a `NO_PRINTER`, então tem que negar a lista estática, mantendo o valor do back.
(tirei a exclamação no value, e coloquei no `!_hasModelAtList`)


```suggestion
  /** negação explicada e comentada em hasNoTouch() */
  let value = VerifyMethodOnSystemWrapper('hasPrinter');

  if (!value) {
    value = !_hasModelAtList(NO_PRINTER);
  }

  return value;
```

Reply:
Tive que colocar `typeof value === 'undefined'`, porque se o método do back retornar `false`, o `if(!value)`, vai fazer cair na lista estática, que é o que não queremos.